File: control-plane/app/services/managed_nodes/providers/resolver.rb
# frozen_string_literal: true

module ManagedNodes
  module Providers
    class Resolver
      def self.resolve(slug, client: nil)
        case slug.to_s
        when "hetzner"
          Hetzner.new(client: client)
        when "digitalocean"
          DigitalOcean.new(client: client)
        else
          raise ArgumentError, "unsupported managed node provider #{slug.inspect}"
        end
      end
    end
  end
end

File: cli/internal/solo/providers/resolver.go
package providers

import "fmt"

func Resolve(slug string) (Provider, error) {
	return ResolveWithToken(slug, "")
}

func ResolveWithToken(slug, token string) (Provider, error) {
	switch slug {
	case "hetzner":
		return NewHetzner(token), nil
	default:
		return nil, fmt.Errorf("unsupported solo server provider %q", slug)
	}
}
